runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/quarkus/CassandraClientProducer.java

```diff
@@ -58,7 +61,8 @@ public QuarkusCqlSessionState quarkusCqlSessionState() {
   @Produces
   @ApplicationScoped
   @Unremovable
-  public QuarkusCqlSession createCassandraClient(QuarkusCqlSessionState quarkusCqlSessionState) {
+  public CompletionStage<QuarkusCqlSession> createCompletionStageOfCassandraClient(
```

Comment:
It seems that injecting of the CompletionStage works without a problem, I think that we need to document 4 cases:
1. client's code is setting `eager-session-init=false` and injecting the only CompletionStage<QuarkusCqlSession> - in that case, there should be no problem with blocking on Vert.x thread.
2. client's code is setting `eager-session-init=false` and injecting `QuarkusCqlSession` - this is problematic and should be avoided. We can detect that and log a warning, see https://github.com/datastax/cassandra-quarkus/pull/127/files#diff-535ef3e00080110cf5c9d6d562874ff1R92-R94
3.  client's code is setting `eager-session-init=true` - injecting `QuarkusCqlSession` AND `CompletionStage<QuarkusCqlSession>` is allowed

Reply:
1. I think a warning is a bit too much. This is not always a problem, it's only a problem afaict if you use Vert.x routes. But if you use regular REST controllers, it's not a problem.
2. Should we make eager-session-init false by default?

Reply:
1. You are right, we would need to detect if someone is using vert.x. I will change it to an `info` level
2. I doubt that majority of quarkus users are using everything async.  I think that they will pick the `QuarkusCqlSession` by default. So maybe we could leave it as `true`, but mention in the docs that if they want to make they apps fully async, then they should set `eager-session-init = false` and inject `CompletionStage<QuarkusCqlSession>`, wdyt?

Reply:
I agree with @tomekl007 here (leave the default as eager-session-init = true), but for a _slightly_ different reason.  eager-session-init = false is the one case that's known to produce unreliable results (for Vert.x reactive routes).  There's a decent argument that we should be more defensive by default and avoid this configuration by default while still allowing users to access it easily if they so desire.